pkg/client/unversioned/transport.go

```diff
@@ -41,6 +41,7 @@ func (rt *userAgentRoundTripper) RoundTrip(req *http.Request) (*http.Response, e
 	}
 	req = cloneRequest(req)
 	req.Header.Set("User-Agent", rt.agent)
+	req.Header.Set("Content-Type", "application/json")
```

Comment:
Not sure if we have any requests that POST plain serialized json over SPDY, but I believe SPDY supports setting Content-Type header.